// @dolbyio/dolbyio-rest-apis-client/dist/communications/conference.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.terminate = exports.updatePermissions = exports.setSpatialListenersAudio = exports.sendMessage = exports.kick = exports.invite = exports.createConference = void 0;
const httpHelpers_1 = require("../internal/httpHelpers");
const rtcpMode_1 = require("./types/rtcpMode");
/**
 * Creates a conference.
 *
 * @link https://docs.dolby.io/communications-apis/reference/create-conference
 *
 * @param accessToken Access token to use for authentication.
 * @param options Options to create the conference.
 *
 * @returns A `Conference` object through a `Promise`.
 */
const createConference = (accessToken, options) => __awaiter(void 0, void 0, void 0, function* () {
    const parameters = {
        dolbyVoice: options.dolbyVoice ? options.dolbyVoice : true,
        liveRecording: options.liveRecording ? options.liveRecording : false,
        rtcpMode: options.rtcpMode ? options.rtcpMode : rtcpMode_1.RTCPMode.Average,
    };
    if (options.pinCode)
        parameters['pincode'] = options.pinCode;
    if (options.ttl)
        parameters['ttl'] = options.ttl;
    if (options.videoCodec)
        parameters['videoCodec'] = options.videoCodec;
    const body = {
        ownerExternalId: options.ownerExternalId,
        parameters: parameters,
    };
    if (options.alias)
        body['alias'] = options.alias;
    if (options.participants) {
        const obj_participants = {};
        for (let index = 0; index < options.participants.length; index++) {
            const participant = options.participants[index];
            obj_participants[participant.externalId] = {
                permissions: participant.permissions,
                notification: participant.notify,
            };
        }
        body['participants'] = obj_participants;
    }
    const strBody = JSON.stringify(body);
    const requestOptions = {
        hostname: 'api.voxeet.com',
        path: '/v2/conferences/create',
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            Authorization: `${accessToken.token_type} ${accessToken.access_token}`,
        },
        body: strBody,
    };
    const response = yield (0, httpHelpers_1.sendPost)(requestOptions);
    return response;
});
exports.createConference = createConference;
/**
 * Invites participants to an ongoing conference. This API can also be used to generate new conference access tokens for an ongoing conference. If the invite request includes participants that are already in the conference, a new conference access token is not generated and an invitation is not sent.
 *
 * @link https://docs.dolby.io/communications-apis/reference/invite-to-conference
 *
 * @param accessToken Access token to use for authentication.
 * @param conferenceId Identifier of the conference.
 * @param participants List of participants to invite to the conference.
 *
 * @returns The list of user tokens for each participants.
 */
const invite = (accessToken, conferenceId, participants) => __awaiter(void 0, void 0, void 0, function* () {
    const obj_participants = {};
    for (let index = 0; index < participants.length; index++) {
        const participant = participants[index];
        obj_participants[participant.externalId] = {
            permissions: participant.permissions,
            notification: participant.notify,
        };
    }
    const body = JSON.stringify({
        participants: obj_participants,
    });
    const options = {
        hostname: 'api.voxeet.com',
        path: `/v2/conferences/${conferenceId}/invite`,
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            Authorization: `${accessToken.token_type} ${accessToken.access_token}`,
        },
        body,
    };
    const response = yield (0, httpHelpers_1.sendPost)(options);
    return response;
});
exports.invite = invite;
/**
 * Kicks participants from an ongoing conference.
 *
 * @link https://docs.dolby.io/communications-apis/reference/kick-from-conference
 *
 * @param accessToken Access token to use for authentication.
 * @param conferenceId Identifier of the conference.
 * @param externalIds External identifiers of the participants to kick our of the conference.
 */
const kick = (accessToken, conferenceId, externalIds) => __awaiter(void 0, void 0, void 0, function* () {
    const body = JSON.stringify({
        externalIds: externalIds,
    });
    const options = {
        hostname: 'api.voxeet.com',
        path: `/v2/conferences/${conferenceId}/kick`,
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            Authorization: `${accessToken.token_type} ${accessToken.access_token}`,
        },
        body,
    };
    yield (0, httpHelpers_1.sendPost)(options);
});
exports.kick = kick;
/**
 * Sends a message to some or all participants in a conference.
 *
 * @link https://docs.dolby.io/communications-apis/reference/send-message
 *
 * @param accessToken Access token to use for authentication.
 * @param conferenceId Identifier of the conference.
 * @param fromExternalId The external ID of the author of the message.
 * @param toExternalIds A list of external IDs that will receive the message. If empty, the message will be broadcasted to all participants in the conference.
 * @param message The message to send.
 */
const sendMessage = (accessToken, conferenceId, fromExternalId, toExternalIds, message) => __awaiter(void 0, void 0, void 0, function* () {
    const body = {
        from: fromExternalId,
        message: message,
    };
    if (toExternalIds && toExternalIds.length > 0) {
        body['to'] = toExternalIds;
    }
    const options = {
        hostname: 'api.voxeet.com',
        path: `/v2/conferences/${conferenceId}/message`,
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            Authorization: `${accessToken.token_type} ${accessToken.access_token}`,
        },
        body: JSON.stringify(body),
    };
    yield (0, httpHelpers_1.sendPost)(options);
});
exports.sendMessage = sendMessage;
/**
 * Sets the spatial audio scene for all listeners in an ongoing conference. This sets the spatial audio environment, the position and direction for all listeners with the spatialAudio flag enabled. The calls are not cumulative, and each call sets all the spatial listener values. Participants who do not have a position set are muted.
 *
 * @link https://docs.dolby.io/communications-apis/reference/set-spatial-listeners-audio
 *
 * @param accessToken Access token to use for authentication.
 * @param conferenceId Identifier of the conference.
 * @param environment The spatial environment of an application, so the audio renderer understands which directions the application considers forward, up, and right and which units it uses for distance.
 * @param listener The listener's audio position and direction, defined using Cartesian coordinates.
 * @param users The users' audio positions, defined using Cartesian coordinates.
 */
const setSpatialListenersAudio = (accessToken, conferenceId, environment, listener, users) => __awaiter(void 0, void 0, void 0, function* () {
    const body = JSON.stringify({
        environment: environment,
        listener: listener,
        users: users,
    });
    const options = {
        hostname: 'api.voxeet.com',
        path: `/v2/conferences/${conferenceId}/spatial-listeners-audio`,
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            Authorization: `${accessToken.token_type} ${accessToken.access_token}`,
        },
        body,
    };
    yield (0, httpHelpers_1.sendPut)(options);
});
exports.setSpatialListenersAudio = setSpatialListenersAudio;
/**
 * Update permissions for participants in a conference. When a participant's permissions are updated, the new token is sent directly to the SDK. The SDK automatically receives, stores, and manages the new token and a `permissionsUpdated` event is sent.
 *
 * @link https://docs.dolby.io/communications-apis/reference/update-permissions
 *
 * @param accessToken Access token to use for authentication.
 * @param conferenceId Identifier of the conference.
 * @param participants List of the participants with their new permissions.
 *
 * @returns The list of user tokens for each participants.
 */
const updatePermissions = (accessToken, conferenceId, participants) => __awaiter(void 0, void 0, void 0, function* () {
    const obj_participants = {};
    for (let index = 0; index < participants.length; index++) {
        const participant = participants[index];
        obj_participants[participant.externalId] = {
            permissions: participant.permissions,
            notification: participant.notify,
        };
    }
    const body = JSON.stringify({
        participants: obj_participants,
    });
    const options = {
        hostname: 'api.voxeet.com',
        path: `/v2/conferences/${conferenceId}/invite`,
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            Authorization: `${accessToken.token_type} ${accessToken.access_token}`,
        },
        body,
    };
    const response = yield (0, httpHelpers_1.sendPost)(options);
    return response;
});
exports.updatePermissions = updatePermissions;
/**
 * Terminates an ongoing conference and removes all remaining participants from the conference.
 *
 * @link https://docs.dolby.io/communications-apis/reference/terminate-conference
 *
 * @param accessToken Access token to use for authentication.
 * @param conferenceId Identifier of the conference.
 */
const terminate = (accessToken, conferenceId) => __awaiter(void 0, void 0, void 0, function* () {
    const options = {
        hostname: 'api.voxeet.com',
        path: `/v2/conferences/${conferenceId}`,
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            Authorization: `${accessToken.token_type} ${accessToken.access_token}`,
        },
    };
    yield (0, httpHelpers_1.sendDelete)(options);
});
exports.terminate = terminate;
