// @dolbyio/dolbyio-rest-apis-client/dist/communications/types/permission.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Permission = void 0;
var Permission;
(function (Permission) {
    /** Allows a participant to invite participants to a conference. */
    Permission["INVITE"] = "INVITE";
    /** Allows a participant to join a conference */
    Permission["JOIN"] = "JOIN";
    /** Allows a participant to send an audio stream during a conference. */
    Permission["SEND_AUDIO"] = "SEND_AUDIO";
    /** Allows a participant to send a video stream during a conference. */
    Permission["SEND_VIDEO"] = "SEND_VIDEO";
    /** Allows a participant to share their screen during a conference. */
    Permission["SHARE_SCREEN"] = "SHARE_SCREEN";
    /** Allows a participant to share a video during a conference. */
    Permission["SHARE_VIDEO"] = "SHARE_VIDEO";
    /** Allows a participant to share a file during a conference. */
    Permission["SHARE_FILE"] = "SHARE_FILE";
    /** Allows a participant to send a message to other participants during a conference. */
    Permission["SEND_MESSAGE"] = "SEND_MESSAGE";
    /** Allows a participant to record a conference. */
    Permission["RECORD"] = "RECORD";
    /** Allows a participant to stream a conference. */
    Permission["STREAM"] = "STREAM";
    /** Allows a participant to kick other participants from a conference. */
    Permission["KICK"] = "KICK";
    /** Allows a participant to update other participants' permissions. */
    Permission["UPDATE_PERMISSIONS"] = "UPDATE_PERMISSIONS";
})(Permission = exports.Permission || (exports.Permission = {}));
