// @dolbyio/dolbyio-rest-apis-client/dist/media/internal/httpHelpers.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.download = exports.sendDelete = exports.sendPut = exports.sendPost = exports.sendGet = void 0;
const httpHelpers = __importStar(require("../../internal/httpHelpers"));
/**
 * Sends a GET request.
 *
 * @param options Request options.
 *
 * @returns A JSON payload object through a Promise.
 */
const sendGet = (options) => {
    options.headers['Authorization'] = `Bearer ${options.accessToken.access_token}`;
    return httpHelpers.sendGet(options);
};
exports.sendGet = sendGet;
/**
 * Sends a POST request.
 *
 * @param options Request options.
 *
 * @returns A JSON payload object through a Promise.
 */
const sendPost = (options) => {
    options.headers['Authorization'] = `Bearer ${options.accessToken.access_token}`;
    return httpHelpers.sendPost(options);
};
exports.sendPost = sendPost;
/**
 * Sends a PUT request.
 *
 * @param options Request options.
 *
 * @returns A JSON payload object through a Promise.
 */
const sendPut = (options) => {
    options.headers['Authorization'] = `Bearer ${options.accessToken.access_token}`;
    return httpHelpers.sendPut(options);
};
exports.sendPut = sendPut;
/**
 * Sends a DELETE request.
 *
 * @param options Request options.
 *
 * @returns A JSON payload object through a Promise.
 */
const sendDelete = (options) => {
    options.headers['Authorization'] = `Bearer ${options.accessToken.access_token}`;
    return httpHelpers.sendDelete(options);
};
exports.sendDelete = sendDelete;
/**
 * Download a file.
 *
 * @param filepath Where to save the file.
 * @param options Request options.
 *
 * @returns A JSON payload object through a Promise.
 */
const download = (filepath, options) => {
    options.headers['Authorization'] = `Bearer ${options.accessToken.access_token}`;
    return httpHelpers.download(filepath, options);
};
exports.download = download;
