// @dolbyio/dolbyio-rest-apis-client/dist/media/analyze.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getResults = exports.start = void 0;
const httpHelpers_1 = require("./internal/httpHelpers");
/**
 * Starts analyzing to learn about your media.
 *
 * The `input` location of your source media file and `output` location of your Analyze JSON results file are required.
 *
 * This is an asynchronous operation so you will receive a `job_id` to be used to get the job status and result.
 *
 * There are additional optional parameters that can be provided to identify the type of content and additional loudness or validation requirements. See the samples for examples of what requests and responses look like.
 *
 * @link https://docs.dolby.io/media-apis/reference/media-analyze-post
 *
 * Beta API
 * This API is being made available as an early preview.
 * If you have feedback on how you'd like to use the API please reach out to share your feedback with our team.
 * https://dolby.io/contact
 *
 * Content Length
 *
 * Media content with duration less than 2 seconds will not be processed. The API will return an ERROR in this case.
 *
 * @param accessToken Access token to use for authentication.
 * @param jobContent Content of the job description as a JSON payload. You can find the definition at this URL: https://docs.dolby.io/media-apis/reference/media-analyze-post
 *
 * @returns The job identifier through a `Promise`.
 */
const start = (accessToken, jobContent) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        hostname: 'api.dolby.com',
        path: '/media/analyze',
        headers: {},
        accessToken,
        body: jobContent,
    };
    const response = yield (0, httpHelpers_1.sendPost)(requestOptions);
    if (response.hasOwnProperty('job_id')) {
        return response['job_id'];
    }
    return null;
});
exports.start = start;
/**
 * Gets Analyze Status.
 *
 * For a given job_id, this method will check the job status.
 *
 * When `status==Success` you'll be able to retrieve your result from the output location you provided in the original POST.
 *
 * @link https://docs.dolby.io/media-apis/reference/media-analyze-get
 *
 * @param accessToken Access token to use for authentication.
 * @param jobId Identifier of the job to retrieve.
 *
 * @returns The `AnalyzeJob` object through a `Promise`.
 */
const getResults = (accessToken, jobId) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        hostname: 'api.dolby.com',
        path: '/media/analyze',
        params: {
            job_id: jobId,
        },
        headers: {},
        accessToken,
    };
    const response = yield (0, httpHelpers_1.sendGet)(requestOptions);
    return response;
});
exports.getResults = getResults;
