// @dolbyio/dolbyio-rest-apis-client/dist/media/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webhooks = exports.platform = exports.transcode = exports.mastering = exports.jobs = exports.io = exports.enhance = exports.diagnose = exports.analyzeSpeech = exports.analyze = void 0;
const analyze = __importStar(require("./analyze"));
exports.analyze = analyze;
const analyzeSpeech = __importStar(require("./analyzeSpeech"));
exports.analyzeSpeech = analyzeSpeech;
const diagnose = __importStar(require("./diagnose"));
exports.diagnose = diagnose;
const enhance = __importStar(require("./enhance"));
exports.enhance = enhance;
const io = __importStar(require("./io"));
exports.io = io;
const jobs = __importStar(require("./jobs"));
exports.jobs = jobs;
const mastering = __importStar(require("./mastering"));
exports.mastering = mastering;
const transcode = __importStar(require("./transcode"));
exports.transcode = transcode;
const platform = __importStar(require("./platform"));
exports.platform = platform;
const webhooks = __importStar(require("./webhooks"));
exports.webhooks = webhooks;
