// @dolbyio/dolbyio-rest-apis-client/dist/media/mastering.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getResults = exports.start = exports.getPreviewResults = exports.startPreview = void 0;
const httpHelpers_1 = require("./internal/httpHelpers");
/**
 * Starts mastering preview to improve your music.
 *
 * The `inputs` location for your source media file as well as the `outputs` location for the processed media file are required.
 *
 * A `preset` applies dynamic EQ processing to shape your music to match a desired sound. There are also additional optional parameters that can be provided to control the mastering output.
 *
 * A `segment` object specifying preview `start` may optionally be provided.
 *
 * This is an asynchronous operation. You receive a `job_id` that you use to retrieve the results when the mastering is complete.
 *
 * To learn more, see the example requests and responses.
 *
 * @link https://docs.dolby.io/media-apis/reference/media-music-mastering-post
 *
 * @param accessToken Access token to use for authentication.
 * @param jobContent Content of the job description as a JSON payload. You can find the definition at this URL: https://docs.dolby.io/media-apis/reference/media-music-mastering-post
 *
 * @returns The job identifier through a `Promise`.
 */
const startPreview = (accessToken, jobContent) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        hostname: 'api.dolby.com',
        path: '/media/master/preview',
        headers: {},
        accessToken,
        body: jobContent,
    };
    const response = yield (0, httpHelpers_1.sendPost)(requestOptions);
    if (response.hasOwnProperty('job_id')) {
        return response['job_id'];
    }
    return null;
});
exports.startPreview = startPreview;
/**
 * Gets Mastering Preview Results
 *
 * For a given job_id, this method will check if the mastering task has completed.
 * When the status is Success you'll be able to retrieve your results from the outputs locations you provided in the original POST.
 *
 * @link https://docs.dolby.io/media-apis/reference/media-music-mastering-preview-get
 *
 * @param accessToken Access token to use for authentication.
 * @param jobId Identifier of the job to retrieve.
 *
 * @returns The `MasteringPreviewJob` object through a `Promise`.
 */
const getPreviewResults = (accessToken, jobId) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        hostname: 'api.dolby.com',
        path: '/media/master/preview',
        params: {
            job_id: jobId,
        },
        headers: {},
        accessToken,
    };
    const response = yield (0, httpHelpers_1.sendGet)(requestOptions);
    return response;
});
exports.getPreviewResults = getPreviewResults;
/**
 * Starts mastering to improve your music.
 *
 * The `inputs` location for your source media file as well as the `outputs` location for the processed media file are required.
 *
 * A `preset` applies dynamic EQ processing to shape your music to match a desired sound. There are also additional optional parameters that can be provided to control the mastering output.
 *
 * This is an asynchronous operation. You receive a `job_id` that you use to retrieve the results when the mastering is complete.
 *
 * To learn more, see the example requests and responses.
 *
 * @link https://docs.dolby.io/media-apis/reference/media-music-mastering-post
 *
 * @param accessToken Access token to use for authentication.
 * @param jobContent Content of the job description as a JSON payload. You can find the definition at this URL: https://docs.dolby.io/media-apis/reference/media-music-mastering-post
 *
 * @returns The job identifier through a `Promise`.
 */
const start = (accessToken, jobContent) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        hostname: 'api.dolby.com',
        path: '/media/master',
        headers: {},
        accessToken,
        body: jobContent,
    };
    const response = yield (0, httpHelpers_1.sendPost)(requestOptions);
    if (response.hasOwnProperty('job_id')) {
        return response['job_id'];
    }
    return null;
});
exports.start = start;
/**
 * Gets Mastering Results
 *
 * For a given job_id, this method will check if the mastering task has completed.
 * When the status is Success you'll be able to retrieve your results from the outputs locations you provided in the original POST.
 *
 * @link https://docs.dolby.io/media-apis/reference/media-music-mastering-get
 *
 * @param accessToken Access token to use for authentication.
 * @param jobId Identifier of the job to retrieve.
 *
 * @returns The `MasteringJob` object through a `Promise`.
 */
const getResults = (accessToken, jobId) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        hostname: 'api.dolby.com',
        path: '/media/master',
        params: {
            job_id: jobId,
        },
        headers: {},
        accessToken,
    };
    const response = yield (0, httpHelpers_1.sendGet)(requestOptions);
    return response;
});
exports.getResults = getResults;
