// @dolbyio/dolbyio-rest-apis-client/dist/media/transcode.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getResults = exports.start = void 0;
const httpHelpers_1 = require("./internal/httpHelpers");
/**
 * Starts Transcoding.
 *
 * Start transcoding to modify the size, bitrates, and formats for your media.
 *
 * @link https://docs.dolby.io/media-apis/reference/media-transcode-post
 *
 * @param accessToken Access token to use for authentication.
 * @param jobContent Content of the job description as a JSON payload. You can find the definition at this URL: https://docs.dolby.io/media-apis/reference/media-transcode-post
 *
 * @returns The job identifier through a `Promise`.
 */
const start = (accessToken, jobContent) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        hostname: 'api.dolby.com',
        path: '/media/transcode',
        headers: {},
        accessToken,
        body: jobContent,
    };
    const response = yield (0, httpHelpers_1.sendPost)(requestOptions);
    if (response.hasOwnProperty('job_id')) {
        return response['job_id'];
    }
    return null;
});
exports.start = start;
/**
 * Gets Transcode Results.
 *
 * For a given job_id, this method will check if the processing task has completed and return the transcode results.
 *
 * When the status is Success you'll be able to retrieve your result from the output location you provided in the original POST.
 *
 * @link https://docs.dolby.io/media-apis/reference/media-transcode-get
 *
 * @param accessToken Access token to use for authentication.
 * @param jobId Identifier of the job to retrieve.
 *
 * @returns The `TranscodeJob` object through a `Promise`.
 */
const getResults = (accessToken, jobId) => __awaiter(void 0, void 0, void 0, function* () {
    const requestOptions = {
        hostname: 'api.dolby.com',
        path: '/media/transcode',
        params: {
            job_id: jobId,
        },
        headers: {},
        accessToken,
    };
    const response = yield (0, httpHelpers_1.sendGet)(requestOptions);
    return response;
});
exports.getResults = getResults;
